packages/flutter/lib/src/services/message_codecs.dart

```diff
@@ -386,9 +386,28 @@ class StandardMessageCodec implements MessageCodec<Object?> {
       }
     } else if (value is String) {
       buffer.putUint8(_valueString);
-      final Uint8List bytes = utf8.encoder.convert(value);
-      writeSize(buffer, bytes.length);
-      buffer.putUint8List(bytes);
+      final Uint8List asciiBytes = Uint8List(value.length);
+      Uint8List? utf8Bytes;
+      int utf8Offset = 0;
+      // Only do utf8 encoding if we encounter non-ascii characters.
+      for(int i = 0; i < value.length; ++i) {
```

Comment:
```suggestion
      for (int i = 0; i < value.length; i += 1) {
```

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-over

Reply:
Oops, it didn't add the suggestion, now it's done.